language/move-prover/bytecode/tests/memory_instr/basic_test.exp

```diff
@@ -401,64 +409,130 @@ fun TestPackref::test8($t0|b: bool, $t1|n: u64, $t2|r_ref: &mut TestPackref::R)
      var $t11: u64
      var $t12: u64
-     var $t13: bool
-     var $t14: u64
-     var $t15: u64
-     var $t16: u64
+     var $t13: u64
+     var $t14: bool
+     var $t15: &mut TestPackref::R
+     var $t16: &mut TestPackref::R
+     var $t17: u64
+     var $t18: u64
+     var $t19: u64
+     var $t20: bool
+     var $t21: bool
+     var $t22: bool
+     var $t23: bool
+     var $t24: bool
+     var $t25: bool
+     var $t26: bool
+     var $t27: bool
+     var $t28: bool
   0: $t6 := 3
   1: $t3 := pack TestPackref::R($t6)
   2: $t7 := 4
   3: $t4 := pack TestPackref::R($t7)
-  4: $t5 := borrow_local($t4)
-  5: label L7
-  6: $t8 := 0
-  7: $t9 := <($t8, $t1)
-  8: if ($t9) goto 9 else goto 29
-  9: label L0
- 10: write_back[LocalRoot($t3)@]($t5)
- 11: trace_local[r1]($t3)
- 12: write_back[LocalRoot($t4)@]($t5)
- 13: trace_local[r2]($t4)
- 14: destroy($t5)
- 15: $t10 := 2
- 16: $t11 := /($t1, $t10)
- 17: $t12 := 0
- 18: $t13 := ==($t11, $t12)
- 19: if ($t13) goto 20 else goto 23
- 20: label L3
- 21: $t5 := borrow_local($t3)
- 22: goto 25
- 23: label L5
- 24: $t5 := borrow_local($t4)
- 25: label L6
- 26: $t14 := 1
- 27: $t1 := -($t1, $t14)
- 28: goto 5
- 29: label L2
- 30: if ($t0) goto 31 else goto 40
- 31: label L8
- 32: write_back[LocalRoot($t3)@]($t5)
- 33: trace_local[r1]($t3)
- 34: write_back[LocalRoot($t4)@]($t5)
- 35: trace_local[r2]($t4)
- 36: destroy($t5)
- 37: $t15 := 0
- 38: TestPackref::test3($t2, $t15)
- 39: goto 48
- 40: label L10
- 41: destroy($t2)
- 42: $t16 := 0
- 43: TestPackref::test3($t5, $t16)
- 44: write_back[LocalRoot($t3)@]($t5)
- 45: trace_local[r1]($t3)
- 46: write_back[LocalRoot($t4)@]($t5)
- 47: trace_local[r2]($t4)
- 48: label L11
- 49: trace_local[r_ref]($t2)
- 50: return ()
+  4: $t8 := borrow_local($t4)
+  5: $t5 := $t8
+  6: label L7
+  7: $t9 := 0
+  8: $t10 := <($t9, $t1)
+  9: if ($t10) goto 10 else goto 49
+ 10: label L0
+ 11: $t20 := is_parent[Reference($t8)@]($t5)
+ 12: if ($t20) goto 13 else goto 17
+ 13: label L12
+ 14: write_back[Reference($t8)@]($t5)
+ 15: write_back[LocalRoot($t4)@]($t8)
+ 16: trace_local[r2]($t4)
+ 17: label L13
+ 18: $t21 := is_parent[Reference($t15)@]($t5)
+ 19: if ($t21) goto 20 else goto 24
+ 20: label L14
+ 21: write_back[Reference($t15)@]($t5)
+ 22: write_back[LocalRoot($t3)@]($t15)
+ 23: trace_local[r1]($t3)
+ 24: label L15
+ 25: $t22 := is_parent[Reference($t16)@]($t5)
+ 26: if ($t22) goto 27 else goto 31
+ 27: label L16
+ 28: write_back[Reference($t16)@]($t5)
+ 29: write_back[LocalRoot($t4)@]($t16)
+ 30: trace_local[r2]($t4)
+ 31: label L17
+ 32: destroy($t5)
+ 33: $t11 := 2
+ 34: $t12 := /($t1, $t11)
+ 35: $t13 := 0
+ 36: $t14 := ==($t12, $t13)
+ 37: if ($t14) goto 38 else goto 42
+ 38: label L3
+ 39: $t15 := borrow_local($t3)
+ 40: $t5 := $t15
+ 41: goto 45
+ 42: label L5
+ 43: $t16 := borrow_local($t4)
+ 44: $t5 := $t16
+ 45: label L6
+ 46: $t17 := 1
+ 47: $t1 := -($t1, $t17)
+ 48: goto 6
+ 49: label L2
+ 50: if ($t0) goto 51 else goto 77
+ 51: label L8
```

Comment:
Wow thats a lot more code

Reply:
Yes, it looks rather suspicious... The duplication is even more obvious in the [CFG](https://github.com/move-language/move/files/9214035/g.pdf).

I have a guess on the reason, but need some more time to verify it and fix it.

Reply:
The reason for the code explosion is explained below.

And I fixed the `clean_and_optimize` pass such that most of the code is optimized away in the next phase, as seen in the 
[CFG](https://github.com/move-language/move/files/9222771/a-8.pdf) here